spec/config/initializers/upload_types_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+REQUIRED_TABLES = [
+  AccreditationAction.name,
+  AccreditationInstituteCampus.name,
+  AccreditationRecord.name,
+  ArfGiBill.name,
+  Complaint.name,
+  Crosswalk.name,
+  EightKey.name,
+  Hcm.name,
+  IpedsHd.name,
+  IpedsIcAy.name,
+  IpedsIcPy.name,
+  IpedsIc.name,
+  Mou.name,
+  Outcome.name,
+  Scorecard.name,
+  Sec702.name,
+  Sva.name,
+  Vsoc.name,
+  Weam.name,
+  IpedsCipCode.name,
+  StemCipCode.name,
+  YellowRibbonProgramSource.name,
+  Sec109ClosedSchool.name
+].freeze
+
+NO_PROD_TABLES = [].freeze
+
+RSpec.describe 'UPLOAD_TYPES' do
+  describe 'all_tables' do
+    it 'lengths should be equal' do
+      expect(UPLOAD_TYPES_ALL_NAMES.length).to eq(UPLOAD_TYPES.length)
+    end
+  end
+
+  describe 'required_table_names' do
+    it 'contains tables' do
+      expect(UPLOAD_TYPES_REQUIRED_NAMES).to eq(REQUIRED_TABLES)
+    end
+  end
+
+  describe 'no_prod_names' do
+    it 'contains tables' do
+      expect(UPLOAD_TYPES_NO_PROD_NAMES).to eq(NO_PROD_TABLES)
```

Comment:
instead of defining a constant for this, could this just be `expect(UPLOAD_TYPES_NO_PROD_NAMES).to be_empty`?

Reply:
In https://github.com/department-of-veterans-affairs/gibct-data-service/pull/751/files the `NO_PROD_TABLES` will have a value in it

Reply:
What does `NO_PROD_TABLES` represent?  Should the list be in code somewhere instead of as a constant in the test?  This seems like it might trip someone up in the future if they don't know about this list to update when they do add "prod tables"?

Reply:
If an object in the array has `not_prod_ready?` set to true then at https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/views/dashboards/_latest_upload.html.erb#L1 if will hide the upload type from being visible in production environment. 

This repo does not use feature flags as they are not implemented. 
There is occasionally a need to hide an upload type from production while it is being tested in staging.

Reply:
What if the test looked something like this?
```
describe 'no_prod_names' do
    it 'contains tables' do
      UPLOAD_TYPES.each do |upload_type|
        if upload_type[:not_prod_ready?]
          expect(UPLOAD_TYPES_NO_PROD_NAMES).to include(upload_type)
        else
          expect(UPLOAD_TYPES_NO_PROD_NAMES).to_not include(upload_type)
        end
      end
    end
  end
```
My concern is that a future dev would set `:not_prod_ready?` to `true` then the test would fail, and to pass it they would have to change the test put it in `NO_PROD_TABLES`, then when it is prod ready the dev making that update would have to change the test again.  So something like I proposed is testing the functionality and not adding a maintenance task.

Reply:
changed both unit tests to take this approach for checking the subsets